spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/bigquery/GcpBigQueryAutoConfiguration.java

```diff
@@ -141,4 +160,16 @@ public BigQueryTemplate bigQueryTemplate(
     return new BigQueryTemplate(
         bigQuery, bigQueryWriteClient, bqInitSettings, bigQueryThreadPoolTaskScheduler);
   }
+
+  private String resolveToHost(String endpoint) throws URISyntaxException {
+    if (!endpoint.contains("://")) {
+      endpoint = "https://" + endpoint;
+    }
+    URI uri = new URI(endpoint);
+
+    // Construct the new URL with https:// and no port
```

Comment:
Can we just do String manipulation is host is basically endpoint without ports? Is there any edge cases can not be handled by String manipulation?

Reply:
Done. Simplified to use String manipulation.